lib/modules/datasource/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  DescribeAddonVersionsCommand,
+  DescribeAddonVersionsCommandInput,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
+import { cache } from '../../../util/cache/package/decorator';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { EKSAddonsFilter, EKSAddonsFilterSchema } from './schema';
+
+export class AwsEKSAddonDataSource extends Datasource {
+  static readonly id = 'aws-eks-addon';
+
+  override readonly caching = true;
+  private readonly eksClients: Record<string, EKSClient> = {};
+
+  constructor() {
+    super(AwsEKSAddonDataSource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${AwsEKSAddonDataSource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `getReleases:${packageName}`,
+  })
+  async getReleases({
+    packageName: serializedFilter,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const filter: EKSAddonsFilter = EKSAddonsFilterSchema.parse(
+      JSON.parse(serializedFilter),
+    );
```

Comment:
```suggestion
    const filter = EksAddonsFilterJson.safeParse(serializedFilter);
```
Simplify and validate result